proposals/type-classes.md

```diff
@@ -39,65 +38,34 @@ In the implementation below we provide evidence that there is a `Monoid<Int>` in
 ```kotlin
 package intext
 
-extension object IntMonoid : Monoid<Int> {
+extension object : Monoid<Int> {
     fun Int.combine(b: Int): Int = this + b
     val empty: Int = 0
 }
 ```
 
-```
-import intext.IntMonoid
-
-1.combine(2) // 3
-IntMonoid.empty() // 0
-```
-
-Because of this constrain where we are stating that there is a `Monoid` constrain for a given type `A` we can also encode polymorphic definitions based on those constrains:
-
-```kotlin
-import intext.IntMonoid
-
-fun <A> add(a: A, b: A, with Monoid<A>): A = a.combine(b)
-add(1, 1) // compiles
-add("a", "b") // does not compile: No `String: Monoid` instance defined in scope
-```
-
-## Compile Time Dependency Injection
-
-On top of the value this brings to typed FP in Kotlin it also helps in OOP contexts where dependencies can be provided at compile time:
-
-```kotlin
-extension interface Context<A> {
-  fun A.config(): Config
-}
-```
-
+Type class implementations can be given a name for Java interop.
 ```kotlin
-package prod
+package intext
 
-extension object ProdContext: Context<Service> {
-  fun Service.config(): Config = ProdConfig
+extension object IntMonoid : Monoid<Int> {
```

Comment:
Are we optionally allowing names for type class interfaces?

Reply:
I think that makes sense since Kotlin users won't be referring to them by name. For Java interop we obviously want to allow naming though.